lib/dao.js

```diff
@@ -324,82 +324,72 @@ DataAccessObject.create = function(data, options, cb) {
-            if (err) return cb(err);
+    var _idName = idName(Model);
+    var modelName = Model.modelName;
+    var val = removeUndefined(obj.toObject(true));
+    function createCallback(err, id, rev) {
+      if (id) {
+        obj.__data[_idName] = id;
+        defineReadonlyProp(obj, _idName, id);
+      }
+      if (rev) {
+        obj._rev = rev;
+      }
+      if (err) {
+        return cb(err, obj);
+      }
+      obj.__persisted = true;
 
-            // By default, the instance passed to create callback is NOT updated
-            // with the changes made through persist/loaded hooks. To preserve
-            // backwards compatibility, we introduced a new setting updateOnLoad,
-            // which if set, will apply these changes to the model instance too.
-            if (Model.settings.updateOnLoad) {
-              obj.setAttributes(context.data);
-            }
-            saveDone.call(obj, function() {
-              createDone.call(obj, function() {
-                if (err) {
-                  return cb(err, obj);
-                }
-                var context = {
-                  Model: Model,
-                  instance: obj,
-                  isNewInstance: true,
-                  hookState: hookState,
-                  options: options,
-                };
+      var context = {
+        Model: Model,
+        data: val,
+        isNewInstance: true,
+        hookState: hookState,
+        options: options,
+      };
+      Model.notifyObserversOf('loaded', context, function(err) {
+        if (err) return cb(err);
 
-                Model.notifyObserversOf('after save', context, function(err) {
-                  cb(err, obj);
-                  if (!err) Model.emit('changed', obj);
-                });
-              });
-            });
-          });
+        // By default, the instance passed to create callback is NOT updated
+        // with the changes made through persist/loaded hooks. To preserve
+        // backwards compatibility, we introduced a new setting updateOnLoad,
+        // which if set, will apply these changes to the model instance too.
+        if (Model.settings.updateOnLoad) {
+          obj.setAttributes(context.data);
         }
 
-        context = {
+        var context = {
           Model: Model,
-          data: val,
+          instance: obj,
           isNewInstance: true,
-          currentInstance: obj,
           hookState: hookState,
           options: options,
         };
-        Model.notifyObserversOf('persist', context, function(err) {
-          if (err) return cb(err);
 
-          if (connector.create.length === 4) {
-            connector.create(modelName, obj.constructor._forDB(context.data), options, createCallback);
-          } else {
-            connector.create(modelName, obj.constructor._forDB(context.data), createCallback);
-          }
+        Model.notifyObserversOf('after save', context, function(err) {
+          if (err) return cb(err);
```

Comment:
I think this line is not necessary, because the only other line in this function is the next one calling `cb(err, obj)`.